### DISCUSSION TITLE
`app` vs `app_name` vs `service.name` vs `job`

### DISCUSSION THREAD

**Discussion by mxab** [grafana/faro-web-sdk#225] [Q&A] [answered]
Hi,

I'm currently trying to get more familiar with Faro, Loki and so on.

I'm a bit confused what they recommedentation would be when it comes to setting a label for your application.

The grafana agent in your example sets a hardcoded `app` label.
Faro itself is configured with: 
```js
initFaro({
   app: {
     name: "...."
   }
....
})
```

Which adds the `app_name` in the log message and it uses the app name also as the otel `service.name` value.

For `app` and `app_name` are `@grafana/faro-demo-client` eventually which seems very redundant to me.

Is the reason to enfore the `app` label from the agent that you cannot trust the cardinality of the frontends `app_name` ? 

This would mean that for every frontend application I deploy, I would need to add its own agent which can enforce the specific `app`  and idealy I would drop the `app_name` in a pipeline? 

When I look at other examples arround loki, promtail and for example spring boot there is often the label `job` used to specify the application name.

Is there any recommendation or best practice on how to label your application?

Kind Regards

**Comment by codecapitano** [comment, marked as answer]
Hi @mxab 

The initial thinking behind this was that we need a label that is enforced by the receiver to prevent spoofing data.

It's about cases where one Faro app as configured on receiver is receiving telemetry from multiple frontends that might want to identify themselves.

The naming ambiguity is very unfortunate and we should change this to a more explicit naming scheme.

/cc @cedricziel
